v3.1/glfw/util.go

```diff
@@ -1,11 +1,36 @@
 package glfw
 
+//#include <stdlib.h>
 //#include "glfw/include/GLFW/glfw3.h"
 import "C"
 
+import (
+	"reflect"
+	"unsafe"
+)
+
 func glfwbool(b C.int) bool {
 	if b == C.GL_TRUE {
 		return true
 	}
 	return false
 }
+
+func bytes(origin []byte) (pointer *uint8, free func()) {
+	if len(origin) == 0 {
+		panic("Slice of bytes cannot be empty.")
+	}
```

Comment:
I don't see the point of panicking if length of `origin` is 0. In `gl.Strs`, it was needed to handle invalid API usage of providing no strings. But I'm not sure that having it here for `bytes` makes sense. If there's a way in which it's helpful, can you explain it?